dash/development/build_process.py

```diff
@@ -0,0 +1,210 @@
+            try:
+                os.makedirs(self.build_folder)
+            except OSError:
+                logger.exception(
+                    "🚨 having issues manipulating %s", self.build_folder
+                )
+                sys.exit(1)
+
+        if self.__class__.__name__ == "DCC":
+            shutil.copyfile(
+                self._concat(self.main, "assets", "highlight.pack.js"),
+                self._concat(self.build_folder, "highlight.pack.js"),
+            )
+            logger.info("copy the customized highligh js")
+
+        versions = {
+            "version": self.version,
+            "package": self.name.replace(" ", "_").replace("-", "_"),
+        }
+        for name, subfolder, filename, target in self.deps_info:
+            version = self.deps[name]["version"]
+            versions[name.replace("-", "").replace(".", "")] = version
+
+            logger.info("copy npm dependency => %s", filename)
+            ext = "min.js" if "min" in filename.split(".") else "js"
+            target = (
+                target.format(version)
+                if target
+                else "{}@{}.{}".format(name, version, ext)
+            )
+            shutil.copyfile(
+                self._concat(self.npm_modules, name, subfolder, filename),
+                self._concat(self.build_folder, target),
+            )
+
+        logger.info("run `npm run build:webpack`")
+        os.chdir(self.main)
+        run_command_with_process("npm run build:webpack")
+
+        logger.info("generate the `__init__.py` from template and verisons")
+        with open(self._concat(self.main, "init.template")) as fp:
+            t = string.Template(fp.read())
+
+        with open(self._concat(self.build_folder, "__init__.py"), "w") as fp:
+            fp.write(t.safe_substitute(versions))
+
+
+class Renderer(BuildProcess):
+    def __init__(self):
+        # dash-renderer's path is binding with the dash folder hierarchy
+        super(Renderer, self).__init__(
+            self._concat(
+                os.path.abspath(__file__),
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                "dash-renderer",
+            ),
+            (
+                ("react", "umd", "react.production.min.js", None),
+                ("react", "umd", "react.development.js", None),
+                ("react-dom", "umd", "react-dom.production.min.js", None),
+                ("react-dom", "umd", "react-dom.development.js", None),
+                ("prop-types", None, "prop-types.min.js", None),
+                ("prop-types", None, "prop-types.js", None),
+            ),
+        )
+
+
+class DCC(BuildProcess):
+    def __init__(self, main=None):
+        main = (
+            os.path.abspath(os.path.expanduser(main)) if main else os.getcwd()
+        )
+        logger.info("start the build process at %s", main)
+        if os.path.basename(main) != "dash-core-components":
```

Comment:
this extra protection might make people changed the folder name unpleasant,  like someone git clone and checkout to `dcc`